Automattic/pocket-casts-android
# Listening history - Display search bar

## Description
This adds a search bar on the listening history screen.

## Testing Instructions


1. Login with an account with sufficient listening history
2. Go to `Profile` tab -> `Listening History`
3. ✅ Notice that a search bar now exists at the top
4. Long press an episode
5. ✅ Notice that multi-select bar is displayed at the top 
6. Dismiss the multi-select toolbar
7. Scroll the listening history list 
8. ✅ Notice that search bar scrolls behind the toolbar
9. Go to `Profile` tab -> `Starred`
10. ✅ Notice that search bar is not present on the screen
11. Go to `Profile` tab -> `Downloads`
12. ✅ Notice that search bar is not present on the screen
13. Disable feature flag
14. Go to `Profile` tab -> `Listening History`
15. ✅ Notice that search bar is not present on the screen

## Screenshots or Screencast 

https://github.com/user-attachments/assets/dcf4824d-91aa-49aa-acbf-98727ce1add7

**Themed**

<img width=300 src="https://github.com/user-attachments/assets/a777d62f-d20a-48e9-842b-db07dbc46c37"/>

**Landscape**

<img width=600 src="https://github.com/user-attachments/assets/e5c2ff21-c998-40b8-92c4-b1462a93ea60"/>


## Checklist
- [ ] Any jetpack compose components I added or changed are covered by compose previews (the only component added is a search bar which has a preview available)
 
#### I have tested any UI changes...
